agent/eni/watcher/watcher_common.go

```diff
@@ -0,0 +1,197 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
```

Comment:
this file can just call watcher.go and have a New() function which calls a private function to initialize the watcher that differs between linux and windows. refer to my comment in watcher_windows.go

Reply:
I have changed the file name to watcher.go
We cannot have a common New method as platform specific interface monitor wrapper is passed to the New method. This was probably done to inject mocks of udevwrapper during unit testing.